spec/models/post_spec.rb

```diff
@@ -0,0 +1,37 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  describe 'validations' do
+    it { should validate_presence_of(:title) }
+    it { should validate_length_of(:title).is_at_most(250) }
+    it { should validate_numericality_of(:comments_counter).only_integer.is_greater_than_or_equal_to(0) }
+    it { should validate_numericality_of(:likes_counter).only_integer.is_greater_than_or_equal_to(0) }
+  end
+
+  describe 'associations' do
+    it { should belong_to(:author).class_name('User') }
+    it { should have_many(:comments).dependent(:destroy) }
+    it { should have_many(:likes).dependent(:destroy) }
+  end
+
+  describe '#five_most_recent_comments' do
+    it 'returns the five most recent comments for the post' do
+      user1 = User.create(name: 'John Doe')
+      post1 = Post.create(author: user, title: 'New post')
+      comments = []
+      10.times do |i|
+        comments.push(Comment.create!(user: user1, post: post1, text: "Comment #{i}"))
+      end
+      expect(post.five_most_recent_comments).to eq(comments.last(5).reverse)
+    end
```

Comment:
👍 Good effort on your work! However, it seems there's room for improvement, specifically regarding a test case that doesn't pass:

- [ ] **Reviewing and Updating Test Case:** It appears that the test case is encountering an issue, possibly related to the reference to the variable `user` in line number 20. Please carefully check your code, and if the variable should indeed be `user`, make sure it is defined correctly.
![image](https://github.com/mohamedSabry0/BlogsDir/assets/55840999/897335dd-7cbb-46c7-bb25-d4a0c2b02681)

By addressing this issue, you'll enhance the reliability of your test cases. Keep up the good work and strive for test case accuracy! 🚀🧪